### ISSUE TITLE
please move away from pygtk 2

### ISSUE THREAD

**Issue by leggewie** [pixelb/fslint#147]
fslint has been removed from Debian testing because it uses 7 years old unmaintained pygtk libraries. pygtk is unmaintained upstream. It has not had a release since GNOME 3 was released in 2011 ! The way forward is to port your app to use GObject Introspection bindings.

For more information on GObject Introspection see
[1] https://wiki.gnome.org/Projects/GObjectIntrospection
[2] https://wiki.gnome.org/Projects/PyGObject

See https://bugs.debian.org/885280 for more info

**Comment by pixelb** [comment]
I'll try and get some time for this, but TBH it's unlikely in the short term

**Comment by pixelb** [comment]
I see python2 will be removed from debian unstable after the release of buster,
so we would also have to rewrite in python 3 as well as moving from pygtk to PyGI+GTK

**Comment by frenzymadness** [comment]
This is going to be urgent. Is there any plan to port fslist to Python 3 and the newer GTK?

**Comment by pixelb** [comment]
No immediate plans, sorry

**Comment by hosiet** [comment]
Given the progress of python2 removal happening in Debian and Ubuntu, I am going to have fslint removed from Debian's official repository (and later Ubuntu). This means that fslint will not appear in future Debian releases and very likely disappear from Ubuntu 20.04 LTS. We can always reintroduce it back once the upstream migration to python3 is complete.

**Comment by Seegras** [comment]
can we strip out the GUI support altogether and keep a (python3) command line version?

**Comment by pixelb** [comment]
yes possible, though I think the GUI is by far the main usage of this package

**Comment by smbrannon** [comment]
My suggestion is to try another comparable tool, such as dupeGuru, which is written in Python 3 with Qt5 on Linux and Windows, and Python 3 with native Objective-C and Cocoa on MacOS. It is in active development. 

https://dupeguru.voltaicideas.net/
https://github.com/arsenetar/dupeguru

There are other alternatives as well; depends on what you are looking for. Google is your friend.  ;-)

**Comment by mbnoimi** [comment]
OMG, thanks @smbrannon  `dupeguru` GUI much better than `flsint` specially for big list of duplicates.
I wonder why `fslint` gurus still insist to stick to pygtk 2!!!

**Comment by smbrannon** [comment]
I believe I saw somewhere that it would take too much work for them to move
to pygtk3.

**Comment by jaakristioja** [comment]
Within 30 days, fslint will be removed from Gentoo Linux as well: https://github.com/gentoo/gentoo/commit/327e9abe41efe8ebf9a9f0151c44edd05f09a441

**Comment by mcepl** [comment]
Just going to file request for removal of fslint from openSUSE/Factory.

**Comment by someone-from-userland** [comment]
I'd also like to bump this one, since for a while python 2.7 support was announced to be over and Gentoo maintainers are doing a major cleanup on packages that still depends on pygtk2 and python 2.7. 
fslint is already masked for removal. 
Would be sad to see it go.

**Comment by Doctor-Who** [comment]
> Just going to file request for removal of fslint from openSUSE/Factory.

I saw that, we should propose an alternative such as dupeguru in official repository

**Comment by scottkosty** [comment]
I just wanted to say a *big* thanks to @pixelb for the maintenance of fslint over the years. It's such a great tool!

I ended up on this issue because I realized the package isn't there on Ubuntu 20.04 daily. Porting to newer versions of dependencies (especially with big changes) can be a huge time commitment. I don't blame pixelb for not having the time. Hopefully someone can volunteer to help.

**Comment by RoestVrijStaal** [comment]
I think a better title is "Move far away from Python".

It's the fault of the devs of Python to choose for a craggy release cycle, which enforces other developers to rewrite their Python application. 

This project is a perfect example why the counterargument "There was plenty of time to migrate!!!!" is weak. pixelb had no time to rewrite his application. Neither other developers.

**Comment by didierga** [comment]
I updated my Fedora from Fedora 31 to Fedora 32 and I discovered fslint is no longer available in Fedora repo:

https://bugzilla.redhat.com/show_bug.cgi?id=1777626

**Comment by Seegras** [comment]
For finding same filenames you can do something like
`find . | grep -E $( find . -type f | sed 's,.*/\([^/]*$\),\1,' | sort | uniq -d | tr '\n' '|' )`
(bombs when no duplicate names are present); or use https://github.com/arsenetar/dupeguru with options "Filter Hardness 100", and all everything elsse checked off; but this still finds near-duplicate names where only caps differ (Which might actually not be what you want. "IT" and "It" are two different movies, for instance).

**Comment by pixelb** [comment]
@didierga You can still install the Fedora 31 package like:

dnf install https://kojipkgs.fedoraproject.org/packages/fslint/2.46/7.fc31/noarch/fslint-2.46-7.fc31.noarch.rpm

**Comment by sbliven** [comment]
@pixelb You should consider marking fslint as an archived project if there are no plans to port to python3.

**Comment by sbaragnaus2** [comment]
What a shame! I am almost forced to use another software. Thanks anyway.

**Comment by Cantuccini-Lori** [comment]
Just a hint for everybody looking for a substitute: The project "Czkawka" is a fslint clone and really good. Its in active development and seems very stable and better than fslint ever was!!

https://github.com/qarmin/czkawka/releases/

**Comment by smbrannon** [comment]
> Just a hint for everybody looking for a substitute: The project "Czkawka" is a fslint clone and really good. Its in active development and seems very stable and better than fslint ever was!!
> 
> https://github.com/qarmin/czkawka/releases/

Thanks for the hint!

**Comment by Cantuccini-Lori** [comment]
You are welcome!

> Just a hint for everybody looking for a substitute: The project "Czkawka" is a fslint clone and really good. Its in active development and seems very stable and better than fslint ever was!!

https://github.com/qarmin/czkawka/releases/

**Comment by insaner** [comment]
As mentioned in #172, I have begun work on porting fslint to python3 and gtk3. If anyone can spare some help, that would be greatly appreciated!

https://github.com/insaner/fslint

**Comment by A-Infor** [comment]
> yes possible, though I think the GUI is by far the main usage of this package

Could it use ncurses to have a kind of graphical interfae?
